### PR TITLE
feat(agent): prompt injection defense and confirmation endpoint

### LINKED ISSUE #18: Prompt injection defense and sandwich prompt pattern
## Description
Implement structured prompt defense: system instructions with clear delimiters, instruction hierarchy (system > user > document), canary tokens, and human-in-the-loop gates for write operations.

## Acceptance Criteria
- [ ] System prompt uses sandwich defense (system → user input → system reminder)
- [ ] Canary token embedded in system prompt
- [ ] Write operations (create/modify/delete events) require user confirmation
- [ ] Input regex filter for known injection phrases
- [ ] Structured output via JSON schema for sensitive operations

## File Scope
- `backend/app/agents/prompts.py` — hardened system prompt
- `backend/app/agents/guardrails.py` — input validation, canary check
- `backend/app/agents/calendar_agent.py` — human-in-the-loop gate

## Dependencies
Requires #16 (ReAct agent).

## Parent Feature
Closes part of #4

### PR DESCRIPTION
**Pull request by chakraborty29**
## Summary
- **Input regex guard** (`guardrails.py`): Blocks 9 categories of prompt injection patterns with whitespace normalization to prevent newline bypasses, case-insensitive matching across all patterns
- **Canary token leak detection**: Strips leaked canary tokens from streamed agent output with structured warning logs
- **`/api/chat/confirm` endpoint**: Pydantic-validated confirmation flow with thread ID ownership enforcement (including bare-prefix rejection), ready for #17 to wire up `interrupt_before=["tools"]`
- **41 tests** covering unit + endpoint integration for all guardrail functions, bypass vectors, and edge cases

## Notes
- `interrupt_before=["tools"]` deferred to #17 — LangGraph raises `ValueError` when `tools=[]`
- Sandwich defense and canary token infrastructure already existed from #16 (unchanged)

## Test plan
- [x] `uv run pytest` — 119 tests pass (41 new + 78 existing)
- [x] `uv run mypy .` — no issues
- [x] `uv run pyright` — 0 errors
- [x] `uv run ruff check .` — all checks passed
- [x] Code review: newline bypass, case-sensitivity, and bare-prefix thread_id findings fixed

Closes #18

🤖 Generated with [Claude Code](https://claude.com/claude-code)

## Release Notes

* **New Features**
  * Added confirmation workflow allowing users to approve or decline agent actions before execution
  * Implemented prompt injection detection and blocking to prevent malicious inputs
  * Added canary token protection to detect and prevent sensitive information leakage
  * Enhanced thread ownership validation to ensure secure access to conversation threads

### CODE REVIEW

File: backend/app/agents/router.py [hunk 1]
```diff
@@ -27,13 +31,65 @@ class ChatRequest(BaseModel):
...
     user_id: str,
 ) -> AsyncGenerator[str, None]:
-    """Stream agent response as SSE events."""
+    """Stream agent response as SSE events.
+
+    Uses a trailing buffer to detect canary tokens split across chunks.
+    """
+    canary = settings.canary_token
+    # Hold back the last (len(canary)-1) chars so a split canary is caught
+    buf_size = max(len(canary) - 1, 0) if canary else 0
```
[the remark this answers is not in the document]
Comment (chakraborty29) [reply backend/app/agents/router.py]: This is intentional — the canary token is an **optional** defense-in-depth mechanism, not a required config.

- `config.py` defaults `canary_token` to `""` by design — dev/test environments don't need one
- `prompts.py` uses the same pattern: `if not settings.canary_token: return _INSTRUCTIONS_BASE` — no canary is embedded in the prompt when unconfigured
- The sandwich prompt is the primary defense; canary detection is a secondary layer that activates only when configured
- Failing fast on empty canary would break every environment that doesn't set `CANARY_TOKEN`, which is all of them in dev/CI

When `canary_token` is empty, `buf_size = 0` and no buffering occurs — chunks stream through directly with zero overhead. This is the correct behavior: no canary configured = no canary detection needed = no performance cost.

File: backend/tests/test_users.py [hunk 2]
```diff
@@ -30,6 +30,117 @@ async def client() -> AsyncGenerator[httpx.AsyncClient, None]:
...
+            "/api/users/me",
+            headers={
+                "X-User-Id": "google-123",
+                "X-User-Email": "   ",
+            },
+        )
+        assert response.status_code == 401
+
+    async def test_should_work_with_dependency_override(
+        self, client: httpx.AsyncClient
```
[the remark this answers is not in the document]
Comment (chakraborty29) [reply backend/tests/test_users.py]: False positive — this file is not part of this PR's changes (rebase artifact), and there is no syntax error.

Verified:
- \`ast.parse()\` on \`test_users.py\` returns OK
- All 11 tests in \`test_users.py\` pass (\`uv run pytest tests/test_users.py -v\`)
- There is no duplicate \`TestGetMeEndpoint\` class — only one definition exists

This file was introduced in PR #55 and appeared in the diff context due to a rebase onto main.
